In [1]:
########################### TABLE 1
library(tidyverse)
library(gtsummary) 
library(dplyr) 
library(stringr)
library(lubridate)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   4.0.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


In [2]:
# 1. Load Data
raw_data <- read.csv("data/data.csv")

In [3]:
# 2. Variable Definitions and Data Cleaning
# This step ensures your raw variables match the manuscript's categories
table_s2_raw <- raw_data %>%
  mutate(
    # --- AGE AT DIAGNOSIS ---
    # Calculated from Year of Diagnosis and Year of Birth 
    Age = as.numeric(Anio_diag) - as.numeric(Anio_nac),
    
    # --- SEX --- 
    Sex = factor(Sexo, levels = c("Mujer", "Hombre"), labels = c("Female", "Male")),
    
    # --- EDUCATION LEVEL --- 
    Education = case_when(
      Nivel_educ == "Ninguna" ~ "None / Illiterate",
      Nivel_educ == "Primaria" ~ "Primary",
      Nivel_educ == "Secundaria" ~ "Secondary",
      Nivel_educ == "Superior" ~ "Higher (University/Technical)",
      TRUE ~ "Missing"
    ),
    Education = factor(Education, levels = c("None / Illiterate", "Primary", "Secondary", 
                                            "Higher (University/Technical)", "Missing")),
    
    # --- DIAGNOSTIC ESTABLISHMENT ---
    Establishment = case_when(
      Establecimiento == "MSP" ~ "Ministry of Public Health (MSP)",
      Establecimiento == "Privados no SOLCA" ~ "Private Sector",
      Establecimiento == "IESS" ~ "Social Security (IESS)",
      Establecimiento == "SOLCA" ~ "SOLCA",
      TRUE ~ "Missing"
    ),
    
    # --- TUMOR MULTIPLICITY (The 'By' Variable) --- 
    # 'Survivors' are those with 2 or more tumors 
    Multiplicity = if_else(Num_tumores >= 2, 
                           "Multiple Primaries (Survivors)", 
                           "Single Primary Only"),
    Multiplicity = factor(Multiplicity, levels = c("Multiple Primaries (Survivors)", "Single Primary Only"))
  ) %>%
  filter(Age >= 18) # Include only adults as per methods

In [4]:
# 3. Generate the Table
# This creates the layout with N=215 vs N=4,387 
table_s2_output <- table_s2_raw %>%
  select(Age, Sex, Education, Establishment, Multiplicity) %>%
  tbl_summary(
    by = Multiplicity,
    statistic = list(
      Age ~ "{median} ({p25}, {p75})",
      all_categorical() ~ "{n} ({p}%)"
    ),
    digits = list(all_categorical() ~ c(1, 1)), # Matches 58.6% style
    label = list(
      Age ~ "Age at Diagnosis",
      Sex ~ "Sex",
      Education ~ "Education Level",
      Establishment ~ "Diagnostic Establishment"
    )
  ) %>%
  # Add p-values using specific tests from manuscript 
  add_p(
    test = list(
      Age ~ "wilcox.test",         # Wilcoxon rank sum test 
      all_categorical() ~ "chisq.test" # Pearson’s Chi-squared test 
    ),
    pvalue_fun = function(x) style_pvalue(x, digits = 3)
  ) %>%
  modify_header(label = "**Characteristic**") %>%
  bold_labels()

In [5]:
# 4. Display the Table
table_s2_output

<div id="yefrfkewzr" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
  <style>#yefrfkewzr table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#yefrfkewzr thead, #yefrfkewzr tbody, #yefrfkewzr tfoot, #yefrfkewzr tr, #yefrfkewzr td, #yefrfkewzr th {
  border-style: none;
}

#yefrfkewzr p {
  margin: 0;
  padding: 0;
}

#yefrfkewzr .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  b

<div id="sslpkaoqhp" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
  <style>#sslpkaoqhp table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#sslpkaoqhp thead, #sslpkaoqhp tbody, #sslpkaoqhp tfoot, #sslpkaoqhp tr, #sslpkaoqhp td, #sslpkaoqhp th {
  border-style: none;
}

#sslpkaoqhp p {
  margin: 0;
  padding: 0;
}

#sslpkaoqhp .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#sslpkaoqhp .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#sslpkaoqhp .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#sslpkaoqhp .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#sslpkaoqhp .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#sslpkaoqhp .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#sslpkaoqhp .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#sslpkaoqhp .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#sslpkaoqhp .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#sslpkaoqhp .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#sslpkaoqhp .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#sslpkaoqhp .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#sslpkaoqhp .gt_spanner_row {
  border-bottom-style: hidden;
}

#sslpkaoqhp .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#sslpkaoqhp .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#sslpkaoqhp .gt_from_md > :first-child {
  margin-top: 0;
}

#sslpkaoqhp .gt_from_md > :last-child {
  margin-bottom: 0;
}

#sslpkaoqhp .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#sslpkaoqhp .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#sslpkaoqhp .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#sslpkaoqhp .gt_row_group_first td {
  border-top-width: 2px;
}

#sslpkaoqhp .gt_row_group_first th {
  border-top-width: 2px;
}

#sslpkaoqhp .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#sslpkaoqhp .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#sslpkaoqhp .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#sslpkaoqhp .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#sslpkaoqhp .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#sslpkaoqhp .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#sslpkaoqhp .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#sslpkaoqhp .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#sslpkaoqhp .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#sslpkaoqhp .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#sslpkaoqhp .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#sslpkaoqhp .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#sslpkaoqhp .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#sslpkaoqhp .gt_left {
  text-align: left;
}

#sslpkaoqhp .gt_center {
  text-align: center;
}

#sslpkaoqhp .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#sslpkaoqhp .gt_font_normal {
  font-weight: normal;
}

#sslpkaoqhp .gt_font_bold {
  font-weight: bold;
}

#sslpkaoqhp .gt_font_italic {
  font-style: italic;
}

#sslpkaoqhp .gt_super {
  font-size: 65%;
}

#sslpkaoqhp .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#sslpkaoqhp .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#sslpkaoqhp .gt_indent_1 {
  text-indent: 5px;
}

#sslpkaoqhp .gt_indent_2 {
  text-indent: 10px;
}

#sslpkaoqhp .gt_indent_3 {
  text-indent: 15px;
}

#sslpkaoqhp .gt_indent_4 {
  text-indent: 20px;
}

#sslpkaoqhp .gt_indent_5 {
  text-indent: 25px;
}

#sslpkaoqhp .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#sslpkaoqhp div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
  <table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="label"><span class='gt_from_md'><strong>Characteristic</strong></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="stat_1"><span class='gt_from_md'><strong>Multiple Primaries (Survivors)</strong><br />
N = 215</span><span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;line-height:0;"><sup>1</sup></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="stat_2"><span class='gt_from_md'><strong>Single Primary Only</strong><br />
N = 4,387</span><span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;line-height:0;"><sup>1</sup></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="p.value"><span class='gt_from_md'><strong>p-value</strong></span><span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;line-height:0;"><sup>2</sup></span></th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="label" class="gt_row gt_left" style="font-weight: bold;">Age at Diagnosis</td>
<td headers="stat_1" class="gt_row gt_center">70 (57, 78)</td>
<td headers="stat_2" class="gt_row gt_center">60 (45, 73)</td>
<td headers="p.value" class="gt_row gt_center"><0.001</td></tr>
    <tr><td headers="label" class="gt_row gt_left" style="font-weight: bold;">Sex</td>
<td headers="stat_1" class="gt_row gt_center"><br /></td>
<td headers="stat_2" class="gt_row gt_center"><br /></td>
<td headers="p.value" class="gt_row gt_center">0.006</td></tr>
    <tr><td headers="label" class="gt_row gt_left">    Female</td>
<td headers="stat_1" class="gt_row gt_center">126.0 (58.6%)</td>
<td headers="stat_2" class="gt_row gt_center">2,135.0 (48.7%)</td>
<td headers="p.value" class="gt_row gt_center"><br /></td></tr>
    <tr><td headers="label" class="gt_row gt_left">    Male</td>
<td headers="stat_1" class="gt_row gt_center">89.0 (41.4%)</td>
<td headers="stat_2" class="gt_row gt_center">2,252.0 (51.3%)</td>
<td headers="p.value" class="gt_row gt_center"><br /></td></tr>
    <tr><td headers="label" class="gt_row gt_left" style="font-weight: bold;">Education Level</td>
<td headers="stat_1" class="gt_row gt_center"><br /></td>
<td headers="stat_2" class="gt_row gt_center"><br /></td>
<td headers="p.value" class="gt_row gt_center">0.011</td></tr>
    <tr><td headers="label" class="gt_row gt_left">    None / Illiterate</td>
<td headers="stat_1" class="gt_row gt_center">11.0 (5.1%)</td>
<td headers="stat_2" class="gt_row gt_center">241.0 (5.5%)</td>
<td headers="p.value" class="gt_row gt_center"><br /></td></tr>
    <tr><td headers="label" class="gt_row gt_left">    Primary</td>
<td headers="stat_1" class="gt_row gt_center">84.0 (39.1%)</td>
<td headers="stat_2" class="gt_row gt_center">1,509.0 (34.4%)</td>
<td headers="p.value" class="gt_row gt_center"><br /></td></tr>
    <tr><td headers="label" class="gt_row gt_left">    Secondary</td>
<td headers="stat_1" class="gt_row gt_center">63.0 (29.3%)</td>
<td headers="stat_2" class="gt_row gt_center">1,363.0 (31.1%)</td>
<td headers="p.value" class="gt_row gt_center"><br /></td></tr>
    <tr><td headers="label" class="gt_row gt_left">    Higher (University/Technical)</td>
<td headers="stat_1" class="gt_row gt_center">54.0 (25.1%)</td>
<td headers="stat_2" class="gt_row gt_center">951.0 (21.7%)</td>
<td headers="p.value" class="gt_row gt_center"><br /></td></tr>
    <tr><td headers="label" class="gt_row gt_left">    Missing</td>
<td headers="stat_1" class="gt_row gt_center">3.0 (1.4%)</td>
<td headers="stat_2" class="gt_row gt_center">323.0 (7.4%)</td>
<td headers="p.value" class="gt_row gt_center"><br /></td></tr>
    <tr><td headers="label" class="gt_row gt_left" style="font-weight: bold;">Diagnostic Establishment</td>
<td headers="stat_1" class="gt_row gt_center"><br /></td>
<td headers="stat_2" class="gt_row gt_center"><br /></td>
<td headers="p.value" class="gt_row gt_center"><0.001</td></tr>
    <tr><td headers="label" class="gt_row gt_left">    Ministry of Public Health (MSP)</td>
<td headers="stat_1" class="gt_row gt_center">28.0 (13.0%)</td>
<td headers="stat_2" class="gt_row gt_center">725.0 (16.5%)</td>
<td headers="p.value" class="gt_row gt_center"><br /></td></tr>
    <tr><td headers="label" class="gt_row gt_left">    Missing</td>
<td headers="stat_1" class="gt_row gt_center">3.0 (1.4%)</td>
<td headers="stat_2" class="gt_row gt_center">361.0 (8.2%)</td>
<td headers="p.value" class="gt_row gt_center"><br /></td></tr>
    <tr><td headers="label" class="gt_row gt_left">    Private Sector</td>
<td headers="stat_1" class="gt_row gt_center">64.0 (29.8%)</td>
<td headers="stat_2" class="gt_row gt_center">986.0 (22.5%)</td>
<td headers="p.value" class="gt_row gt_center"><br /></td></tr>
    <tr><td headers="label" class="gt_row gt_left">    Social Security (IESS)</td>
<td headers="stat_1" class="gt_row gt_center">72.0 (33.5%)</td>
<td headers="stat_2" class="gt_row gt_center">1,392.0 (31.7%)</td>
<td headers="p.value" class="gt_row gt_center"><br /></td></tr>
    <tr><td headers="label" class="gt_row gt_left">    SOLCA</td>
<td headers="stat_1" class="gt_row gt_center">48.0 (22.3%)</td>
<td headers="stat_2" class="gt_row gt_center">923.0 (21.0%)</td>
<td headers="p.value" class="gt_row gt_center"><br /></td></tr>
  </tbody>
  <tfoot>
    <tr class="gt_footnotes">
      <td class="gt_footnote" colspan="4"><span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;line-height:0;"><sup>1</sup></span> <span class='gt_from_md'>Median (Q1, Q3); n (%)</span></td>
    </tr>
    <tr class="gt_footnotes">
      <td class="gt_footnote" colspan="4"><span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;line-height:0;"><sup>2</sup></span> <span class='gt_from_md'>Wilcoxon rank sum test; Pearson’s Chi-squared test</span></td>
    </tr>
  </tfoot>
</table>
</div>